### ISSUE TITLE
add support for launching Cygwin from "Open in Shell"

### ISSUE THREAD

**Issue by LordOfTheThunder** [desktop/desktop#5654]
**Provide a step-by-step description of the suggested enhancement**
When going to options there is a predefined list of options for shell to use. As you can see I have 3 options, command prompt, power shell and git bash. My enhancement suggestion is to allow manual selection of a shell. For example, I would like to use my cygwin version to run as a shell by clicking "Repository->Open in Cygwin" so I need to be able to select it in the options.
![image](https://user-images.githubusercontent.com/43252111/45586878-2cf50e80-b906-11e8-89d3-ba341084bffa.png)

**Explain why this enhancement would be useful**
It would be much more comfortable using a shell you are already used to working on instead of the predefined ones. Also there are issues in other shells for me which I was able to fix on cygwin and instead of wasting time on configuring other shells I would just like to easily use my own by click option inside desktop github (right now I just open it outside of desktop github and navigate to the sandbox location manually).
![image](https://user-images.githubusercontent.com/43252111/45586863-f61ef880-b905-11e8-8c84-528b7e410b8b.png)

**Comment by shiftkey** [comment]
@LordOfTheThunder thanks for the feedback!

>  For example, I would like to use my cygwin version to run as a shell by clicking "Repository->Open in Cygwin" so I need to be able to select it in the options.

How about we figured out how to detect you had Cygwin installed and add support for that? That would mean that everyone who uses Desktop and Cygwin can come in and just select it, without having to mess around with menus and finding the right program to launch on disk?

**Comment by LordOfTheThunder** [comment]
Thank you @shiftkey . This sounds like a better solution!

**Comment by shiftkey** [comment]
Some quick notes:

 - We have a [guide](https://github.com/desktop/desktop/blob/master/docs/technical/shell-integration.md#windows) to the shell integration on Windows, which favours checking the registry to find the shells installed by the user

 - Cygwin64 installs by default to `C:\cygwin64`, but you should check the registry at `HKEY_LOCAL_MACHINE\SOFTWARE\Cygwin\setup` for the `rootdir` value which is the correct value.

<img width="794" src="https://user-images.githubusercontent.com/359239/45587249-d540ae00-b8d9-11e8-8f40-ab7099c0dd5b.png">

 - Cygwin32 installs by default to `C:\cygwin`, but you should check the registry at `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Cygwin\setup` for the `rootdir` value which will be correct.

<img width="769" src="https://user-images.githubusercontent.com/359239/45587258-f43f4000-b8d9-11e8-8ffc-6d683955552e.png">

- The default shortcuts are set to launch `${CYGWIN_INSTALL_DIR}\bin\mintty.exe -i /Cygwin-Terminal.ico -` which seems reasonable to also launch from within Desktop

 - I didn't see a command line argument to `mintty.exe` that would represent it's start directory (so that we can open the shell in the user's repository), but Node has `cwd` which will work fine.

I think there's enough detail there for a contributor to pick up and try implementing.

**Comment by LordOfTheThunder** [comment]
Thank you @shiftkey . I was able to manually add cygwin to the code using the guide. Unfortunately I have issues with Yarn and node-sass due to which I am incapable of building the project, testing it and pushing the solution. Will be glad if you could add it for next release.
Thanks in advance

**Comment by shiftkey** [comment]
@LordOfTheThunder are you running Node 10? We're aware of this and looking to do the upgrade in #5644 so that you can build it using Node 8 or Node 10.

**Comment by LordOfTheThunder** [comment]
> @LordOfTheThunder are you running Node 10? We're aware of this and looking to do the upgrade in #5644 so that you can build it using Node 8 or Node 10.
Here's what I run:
$ node -v
v8.11.3

$ yarn -v
1.9.4

$ python --version
Python 2.7.14

I fail at the first yarn. I have the right node so I don't think it's a node issue. If you have any ideas I'll be glad to hear

**Comment by shiftkey** [comment]
> I have the right node so I don't think it's a node issue. If you have any ideas I'll be glad to hear

Then it might be related to native modules and requiring Visual Studio - check out the links in the setup guide for more information: https://github.com/desktop/desktop/blob/master/docs/contributing/setup.md#windows

**Comment by LordOfTheThunder** [comment]
Thanks a lot for your help! It works now!

**Comment by LordOfTheThunder** [comment]
I'm sorry if I'm asking some silly questions. I'm just new to github and it takes some time getting used to.
I finished writing the code and testing and my solution for this works.
I commited and pushed it into my own fork but since I don't have push permissions to desktop there is no way for me to upload the code here.
What should I do for my solution to get into the next release of github desktop?
